cmake/unified_kernels.cmake

```diff
@@ -0,0 +1,77 @@
+function(ginkgo_add_unified_library_internal target_name exec stub)
+    add_library(${target_name}_${exec} STATIC ${ARGN})
+    target_link_libraries(${target_name}_${exec} PUBLIC Ginkgo::ginkgo)
+    if(NOT stub)
+        target_compile_definitions(${target_name}_${exec} PRIVATE GKO_COMPILING_STUB GKO_DEVICE_NAMESPACE=${exec})
+    else()
+        string(TOUPPER ${exec} exec_upper)
+        target_compile_definitions(${target_name}_${exec} PRIVATE GKO_COMPILING_${exec_upper})
+    endif()
+    target_link_libraries(${target_name} PRIVATE ${target_name}_${exec})
+endfunction(ginkgo_add_unified_library_internal)
+
+function(ginkgo_add_unified_kernels target_name)
```

Comment:
Could you add the comments that it also generates reference kernel, so ginkgo internal should not use it?